crates/bvs-vault-bank-tokenized/src/contract.rs

```diff
@@ -0,0 +1,650 @@
+            return Err(VaultError::zero("No queued shares").into());
+        }
+
+        if unlock_timestamp.seconds() > env.block.time.seconds() {
+            return Err(VaultError::locked("The shares are locked").into());
+        }
+
+        let claimed_assets = {
+            let underlying_token_balance = UnderlyingToken::query_balance(&deps.as_ref(), &env)?;
+            let receipt_token_supply =
+                cw20_base::contract::query_token_info(deps.as_ref())?.total_supply;
+            let vault = offset::VirtualOffset::new(receipt_token_supply, underlying_token_balance)?;
+
+            let assets = vault.shares_to_assets(queued_shares)?;
+            if assets.is_zero() {
+                return Err(VaultError::zero("Withdraw assets cannot be zero.").into());
+            }
+
+            assets
+        };
+
+        // bank transfer of asset to msg.recipient
+        let transfer_msg = UnderlyingToken::bank_send(deps.storage, &msg.0, claimed_assets)?;
+
+        // When staker queued the withdrawal
+        // The receipt token is ill-liquidated from the staker
+        // by moving the asset into this vault balance.
+        // So the vault should burn from its own balance for the same amount.
+        let msg_info = MessageInfo {
+            sender: env.contract.address.clone(),
+            funds: vec![],
+        };
+        receipt_token_burn(deps.branch(), env.clone(), msg_info, queued_shares)?;
+
+        let receipt_token_supply =
+            cw20_base::contract::query_token_info(deps.as_ref())?.total_supply;
+
+        // Remove staker's info
+        shares::remove_queued_withdrawal_info(deps.storage, &info.sender);
+
+        Ok(Response::new()
+            .add_event(
+                Event::new("RedeemWithdrawalTo")
+                    .add_attribute("sender", info.sender.to_string())
+                    .add_attribute("recipient", msg.0.to_string())
+                    .add_attribute("sub_shares", queued_shares.to_string())
+                    .add_attribute("claimed_assets", claimed_assets.to_string())
+                    .add_attribute("total_shares", receipt_token_supply.to_string()),
+            )
+            .add_message(transfer_msg))
+    }
+
+    /// Moves the assets from the vault to the `vault-router` contract.
+    /// Part of the [https://build.satlayer.xyz/architecture/slashing](Programmable Slashing) lifecycle.
+    /// This function can only be called by `vault-router`, and takes an absolute `amount` of assets to be moved.
+    /// The amount is calculated and enforced by the router.
+    pub fn slash_locked(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        amount: bvs_vault_base::msg::Amount,
+    ) -> Result<Response, ContractError> {
+        router::assert_router(deps.as_ref().storage, &info)?;
+
+        // if the code get passed above assert_router, it means the sender is the router
```

Comment:
```suggestion
        // If the code above (`assert_router`) succeeds, it means the sender is the router
```

Grammar. Didn't notice this previously